pico/api/src/main/java/io/helidon/pico/spi/BasicInjectionPlan.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.spi;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.helidon.builder.Builder;
+import io.helidon.builder.Singular;
+import io.helidon.pico.InjectionPointInfo;
+import io.helidon.pico.ServiceProvider;
+
+/**
+ * Represents the injection plan targeting a given {@link io.helidon.pico.ServiceProvider}.
+ */
+@Builder
+public interface BasicInjectionPlan {
```

Comment:
This name does not make sense. How is it "Basic"? Is there a "Complex" one?

Reply:
BasicInjectionPlan in the api module is extended in the services module with `InjectionPlan`. Rename Basic->Base?

Reply:
Rename `BasicInjectionPlan` to `InjectionPlan` (this is the API), then rename `InjectionPlan` to `PicoInjectionPlan` or similar (FullInjectionPlan, RuntimeInjectionPlan...) for the services

Reply:
pushed